http://bugs.gentoo.org/185104

fix from upstream for:
Padlock engine fails to compile with -O0 -fPIC [from HEAD]

Index: crypto/engine/eng_padlock.c
===================================================================
RCS file: /usr/local/src/openssl/CVSROOT/openssl/crypto/engine/Attic/eng_padlock.c,v
retrieving revision 1.14.2.6
retrieving revision 1.14.2.7
diff -u -p -r1.14.2.6 -r1.14.2.7
--- crypto/engine/eng_padlock.c	19 Oct 2006 20:56:31 -0000	1.14.2.6
+++ crypto/engine/eng_padlock.c	20 May 2007 07:14:14 -0000	1.14.2.7
@@ -436,7 +436,7 @@ static inline void *name(size_t cnt,		\
 			rep_xcrypt "\n"		\
 		"	popl	%%ebx"		\
 		: "=a"(iv), "=c"(cnt), "=D"(out), "=S"(inp) \
-		: "0"(cdata), "1"(cnt), "2"(out), "3"(inp), "m"(*cdata)  \
+		: "0"(cdata), "1"(cnt), "2"(out), "3"(inp)  \
 		: "edx", "cc", "memory");	\
 	return iv;				\
 }
